src/predict_model.py

```diff
@@ -5,52 +5,83 @@
 import torch
 
 import torch
-from torch.utils.data import DataLoader
-from models.model import DeiTClassifier
+from torchvision import transforms
+
+from src.models.model import DeiTClassifier
+from src.train_model import CustomTensorDataset
 
 import hydra
 import wandb
 
-log = logging.getLogger(__name__)
-
-# os.environ["CUDA_LAUNCH_BLOCKING"] = "1" # For CUDA 10.1
-os.environ["CUBLAS_WORKSPACE_CONFIG"] = ":16:8"  # For CUDA >= 10.2
-# https://docs.nvidia.com/cuda/cublas/index.html#cublasApi_reproducibility
-
-torch.use_deterministic_algorithms(True)
 
-
-@hydra.main(config_path="config", config_name="default_config.yaml")
+@hydra.main(config_path=".", config_name="pred_config.yaml", version_base="1.2")
 def predict(
     config,
-    model: torch.nn.Module,
-    dataloader: torch.utils.data.DataLoader,
-    model_checkpoint: str = "/models/saved_models/model.pt",
+    model_checkpoint: str = "models/saved_models/model.pt",
 ) -> None:
     """
     Run prediction for a given model and dataloader.
+    Run src/data/make_subset.py first.
 
     Args:
-        model (nn.Module): Model to use for prediction.
-        dataloader (data.DataLoader): Dataloader with batches of data.
+        model_checkpoint (str): Saved model to use for prediction.
+        data_path (str): Data used for prediction.
 
-    Returns:
-        torch.Tensor: A tensor of shape [N, d], where N is the number of samples
-                      and d is the output dimension of the model.
     """
 
     config = config["hyperparameters"]
-    torch.manual_seed(config["seed"])
     device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
-
     wandb.init(project=config.project_name, entity=config.user)
-
     logging.basicConfig(level=logging.INFO)
     logger = logging.getLogger(__name__)
     wandb.config = config
 
+    pred_images = torch.load(os.path.join(config["data_path"] + "pred_images.pt"))
+    pred_target = torch.load(config["data_path"] + "pred_target.pt")
+    transform = transforms.Compose(
+        [
+            transforms.RandomHorizontalFlip(p=0.5),  # 50% chance of applying a horizontal flip
+            transforms.RandomRotation(10),  # Rotate the image by up to 10 degrees
+            transforms.RandomResizedCrop(48, scale=(0.8, 1.0)),  # Zoom in on the image
+        ]
+    )
```

Comment:
While doing prediction we dont want to transform images. It is only done on training